tests/js/client/shell/shell-transaction-intermediate-commit-babies-cluster.js

```diff
@@ -0,0 +1,306 @@
+      }
+
+      // follower must not have been dropped
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+      assertEqual(droppedFollowersBefore, droppedFollowersAfter);
+      
+      // no intermediate commit must have happened
+      let intermediateCommitsAfter = getMetric(leader, "arangodb_intermediate_commits_total");
+      assertEqual(intermediateCommitsBefore, intermediateCommitsAfter);
+
+      assertEqual(0, c.count());
+      
+      assertInSync(leader, follower, shardId);
+    },
+    
+    testAqlIntermediateCommitsWithOtherFailurePoint: function () {
+      let c = db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = c.shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      let follower = getEndpointById(shards[shardId][1]);
+      // trigger a certain failure point on the leader
+      debugSetFailAt(leader, "insertLocal::fakeResult2");
+      
+      let droppedFollowersBefore = getMetric(leader, "arangodb_dropped_followers_total");
+      let intermediateCommitsBefore = getMetric(leader, "arangodb_intermediate_commits_total");
+      try {
+        db._query('FOR i IN 1..1000 INSERT { _key: CONCAT("test", i) } IN ' + cn, {}, {intermediateCommitCount: 100});
+        fail();
+      } catch (err) {
+        assertEqual(internal.errors.ERROR_DEBUG.code, err.errorNum);
+      }
+
+      // follower must not have been dropped
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+      assertEqual(droppedFollowersBefore, droppedFollowersAfter);
+      
+      // no intermediate commit must have happened
+      let intermediateCommitsAfter = getMetric(leader, "arangodb_intermediate_commits_total");
+      assertEqual(intermediateCommitsBefore, intermediateCommitsAfter);
+
+      assertEqual(0, c.count());
+      
+      assertInSync(leader, follower, shardId);
+    },
+    
+    testAqlIntermediateCommitsWithRollback: function () {
+      let c = db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = c.shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      let follower = getEndpointById(shards[shardId][1]);
+      
+      let droppedFollowersBefore = getMetric(leader, "arangodb_dropped_followers_total");
+      let intermediateCommitsBefore = getMetric(leader, "arangodb_intermediate_commits_total");
+      let didWork = false;
+      try {
+        db._query('FOR i IN 1..10000 INSERT { _key: CONCAT("test", i), value: ASSERT(i < 10000, "peng!") } IN ' + cn, {}, {intermediateCommitCount: 100});
+        didWork = true;
+      } catch (err) {
+      }
+      assertFalse(didWork);
+      
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
```

Comment:
Why do we expect a dropped follower in this case?

Reply:
Added a comment to the PR.